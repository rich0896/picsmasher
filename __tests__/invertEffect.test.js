const { InvertEffect } = require('../effects.js');
const { createTestCanvas } = require('./helpers/testHelpers.js');

describe('InvertEffect', () => {
    test('should invert colors correctly', async () => {
        const { canvas, ctx } = await createTestCanvas();
        const effect = new InvertEffect({ intensity: 100 });

        const consoleSpy = jest
            .spyOn(console, 'log')
            .mockImplementation(() => {});

        effect.apply(ctx, canvas);

        expect(consoleSpy).toHaveBeenCalledWith('Applying Invert Effect');
        consoleSpy.mockRestore();

        const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
        expect(imageData).toMatchSnapshot();
    });

    test('should invert colors correctly at 50% intensity', async () => {
        const { canvas, ctx } = await createTestCanvas();
        const effect = new InvertEffect({ intensity: 50 });

        effect.apply(ctx, canvas);

        const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
        expect(imageData).toMatchSnapshot();
    });

    test('should handle zero intensity without changes', async () => {
        const { canvas, ctx } = await createTestCanvas();
        const effect = new InvertEffect({ intensity: 0 });

        effect.apply(ctx, canvas);

        const imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
        // Compare with original image data
        const originalImageData = ctx.getImageData(
            0,
            0,
            canvas.width,
            canvas.height
        );
        expect(imageData.data).toEqual(originalImageData.data);
    });
});
