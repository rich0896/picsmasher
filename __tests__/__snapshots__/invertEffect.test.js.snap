// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvertEffect should invert colors correctly at 50% intensity 1`] = `ImageData(10, 10, [128, 128, 128, 255, 128, 128, 128, 255, 128, 128...])`;

exports[`InvertEffect should invert colors correctly at 100% intensity 1`] = `ImageData(10, 10, [155, 105, 55, 255, 155, 105, 55, 255, 155, 105...])`;
